src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -105,6 +105,20 @@ function is(x, y) {
 }
 /*eslint-enable no-self-compare*/
 
+/**
+ * We use an Error-like object for backward compatibility as people may call
+ * PropTypes directly and inspect their output. However we don't use real
+ * Errors anymore. We don't inspect their stack anyway, and creating them
+ * is prohibitively expensive if they are created too often, such as what
+ * happens in oneOfType() for any type before the one that matched.
+ */
+function PropTypeError(message) {
+  this.message = message;
+  this.stack = '';
+}
+// Make `instanceof Error` still work for returned errors.
+PropTypeError.prototype = Error.prototype;
```

Comment:
I think this could be better if we don't care about IE<9:
```
function PropTypeError(message) {
  this.message = message;
}
PropTypeError.prototype = Object.create(Error.prototype, { stack: { value: '' }});
```